server/echoserver.go

```diff
@@ -27,16 +59,15 @@ var (
 )
 
 func main() {
-	kingpin.MustParse(app.Parse(os.Args[1:]))
-	serverPort := *port
-	listener, err := net.Listen("tcp", ":"+serverPort)
-	if err != nil {
-		log.Fatal(err)
-	}
-	http.HandleFunc("/echo", echo)
-	log.Println("Listening on localhost:" + serverPort)
-	err = http.Serve(listener, nil)
+	token, err := generateToken(20)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal("unable to generate token", err)
 	}
+	authToken = token
+	fmt.Println("Authentication token: " + authToken)
+
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	serverPort := *port
+	http.HandleFunc("/echo", isAuthorized(echo))
+	log.Fatal(http.ListenAndServeTLS(":"+serverPort, "localhost.crt", "localhost.key", nil))
```

Comment:
Where are these certificates coming from?  Can you add some information/automation on how to create them... Or just check them into the repo -- I recognize you should *never* check in private keys for certs where we care about the security, but for the purposes of this exercise I think it is probably ok. [precedent](https://github.com/bernardjkim/linux-job-worker/tree/master/sslcert)

Reply:
I'll check them in